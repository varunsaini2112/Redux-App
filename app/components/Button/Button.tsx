import { ForwardRefExoticComponent, RefAttributes, forwardRef } from "react";
import {
	Pressable,
	PressableProps,
	StyleProp,
	View,
	ViewStyle
} from "react-native";
import { baseStyles } from "../../config/appStyles";

const Button: ForwardRefExoticComponent<PressableProps & RefAttributes<View>> =
	forwardRef((props, ref) => {
		const { style, ...rest } = props;

		return (
			<Pressable
				ref={ref}
				style={({ pressed }) => [
					{ opacity: pressed ? 0.6 : 1 },
					baseStyles.button,
					style as StyleProp<ViewStyle>
				]}
				{...rest}
			/>
		);
	});

export default Button;
