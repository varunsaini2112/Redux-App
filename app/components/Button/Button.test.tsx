import { fireEvent, render, screen } from "@testing-library/react-native";
import { Text } from "react-native";

import Button from "./Button";

describe("Button Component", () => {
	it("should match snapshots", () => {
		const snapshot = render(
			<Button>
				<Text>Press me</Text>
			</Button>
		);

		expect(snapshot).toMatchSnapshot();
	});

	it("should render text", () => {
		render(
			<Button>
				<Text>Press me</Text>
			</Button>
		);

		expect(screen.getByText("Press me")).toBeOnTheScreen();
	});

	it("onPress should work on button tap", () => {
		const onPressMock = jest.fn();
		render(
			<Button onPress={onPressMock}>
				<Text>Press me</Text>
			</Button>
		);

		fireEvent.press(screen.getByText("Press me"));

		expect(onPressMock).toHaveBeenCalled();
	});

	it("button style should change onPressed state", () => {
		render(
			<Button testOnly_pressed={true} testID="pressable-button">
				<Text>Press me</Text>
			</Button>
		);

		const button = screen.getByTestId("pressable-button");

		expect(button).toHaveStyle({ opacity: 0.6 });
	});
});
