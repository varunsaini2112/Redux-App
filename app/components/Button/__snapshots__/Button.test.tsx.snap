// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshots 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "opacity": 1,
      },
      {
        "alignItems": "center",
        "backgroundColor": "#D80064",
        "borderRadius": 16,
        "marginVertical": 8,
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <Text>
    Press me
  </Text>
</View>
`;
